lib/buffer.js

```diff
@@ -804,10 +804,14 @@ Buffer.prototype.write = function(string, offset, length, encoding) {
 
 
 Buffer.prototype.toJSON = function() {
-  return {
-    type: 'Buffer',
-    data: Array.prototype.slice.call(this, 0)
-  };
+  if (this.length) {
+    const data = [];
```

Comment:
It is until appropriate backports to V8 5.4 are made. I've just decided to go ahead and use this method. See [this issue](https://github.com/nodejs/node/pull/9643) for more information.